src/main/webapp/components/directives/cache/security/security.js

```diff
@@ -0,0 +1,158 @@
+            }
+          });
+          return roles;
+        };
+
+        scope.isSecurityEnabled = function () {
+          return scope.data['enabled'] === true;
+        };
+
+        scope.undoFieldChange = function (field) {
+          scope.data[field] = scope.prevData[field];
+          scope.metadata[field].uiModified = false;
+          scope.metadata[field].style = null;
+        };
+
+        // toggle selection for a given role by name
+        scope.toggleSelection = function toggleSelection(role) {
+          var idx = scope.data.roles.indexOf(role);
+
+          // is currently selected
+          if (idx > -1) {
+            scope.data.roles.splice(idx, 1);
+          }
+
+          // is newly selected
+          else {
+            scope.data.roles.push(role);
+          }
+        };
+
+        scope.isRoleSelected = function (role) {
+          return scope.data.roles.indexOf(role) > -1;
+        };
+
+        scope.isFieldValueModified = function (field) {
+          return utils.isNotNullOrUndefined(scope.metadata[field]) && scope.metadata[field].uiModified === true;
+        };
+
+        scope.isAnyFieldModified = function () {
+          return scope.isFieldValueModified('enabled');
+        };
+
+
+        scope.fieldValueModified = function (field) {
+          if (scope.prevData[field] !== scope.data[field]) {
+            scope.metadata[field].uiModified = true;
+            scope.metadata[field].style = {'background-color': '#fbeabc'};
+            scope.$emit('configurationFieldDirty', field);
+          } else {
+            scope.$emit('configurationFieldClean', field);
+            scope.metadata[field].uiModified = false;
+            scope.metadata[field].style = null;
+          }
+        };
+
+        scope.cleanFieldMetadata = function (field) {
+          if (utils.isNotNullOrUndefined(scope.metadata[field])){
+            scope.metadata[field].uiModified = false;
+            scope.metadata[field].style = null;
+          } else {
+            console.log("Cleaning metadata for configuration field " + field + ", that field does not exist in DMR model")
+          }
+        };
+
+        scope.cleanMetadata = function () {
+          var fields = ['enabled'];
+          fields.forEach(function (field) {
+            scope.cleanFieldMetadata(field);
+            if (utils.isNotNullOrUndefined(scope.data[field])) {
+              scope.prevData[field] = angular.copy(scope.data[field]);
+            } else {
+              scope.prevData[field] = '';
+            }
+          });
+        };
+
+        scope.undoFieldChange = function (field) {
+          scope.data[field] = scope.prevData[field];
+          scope.metadata[field].uiModified = false;
+          scope.metadata[field].style = null;
+        };
+
+        scope.fieldChangeRequiresRestart = function (field) {
+          return utils.isNotNullOrUndefined(scope.metadata[field]) && scope.metadata[field]['restart-required'] !== 'no-services';
```

Comment:
Currently this is not utilised anywhere, but I think it should be based upon my comments below. The template will also need updating to reflect this.